builder/parser/parser_test.go

```diff
@@ -73,3 +73,40 @@ func TestTestData(t *testing.T) {
 		}
 	}
 }
+
+func TestParseWords(t *testing.T) {
+	tests := []map[string][]string{
+		{
+			"input":  {"foo"},
+			"expect": {"foo"},
+		},
+		{
+			"input":  {"foo bar"},
+			"expect": {"foo", "bar"},
+		},
+		{
+			"input":  {"foo=bar"},
+			"expect": {"foo=bar"},
```

Comment:
This test verifies the `parseWords` utility function that I have added to extract the existing word/phrase parsing state-machine from the `parseNameVal` function. I use the same state machine for `parseNameOrNameVal` as described in my comment above. 

Accordingly these are valid tests for `parseWords`. The `ARG` specific tests are part of the `integration-cli`